test/powershell/Modules/Microsoft.PowerShell.Management/Set-Location.Tests.ps1

```diff
@@ -231,3 +231,58 @@ Describe "Set-Location" -Tags "CI" {
         }
     }
 }
+
+Describe "Set-Location: Name with special/wildcards characters" -Tags "CI" {
+
+    BeforeAll {
+
+        $startLocation = (Get-Location).Path
+
+        # printable ascii characters excluding alphanumerics
+        $specialChars =  ' !"#$%&''()*+,-./:;<=>?@[\]^_`{|}~	'.ToCharArray()
+        $testCases = @()
+        ForEach ($specialChar in $specialChars) {
+            If ($specialChar -ne '.') { $testCases += @{Name=([string] $specialChar)} }
+            $testCases += @{Name=('foo' + $specialChar)}
+            $testCases += @{Name=($specialChar + 'foo')}
+        }
+    }
+
+    It "Should be able to create and set location to <Name>" -TestCases $testCases {
+        param ( $Name )
+
+        # skip unsupported characters for the current platform 
+        if ($Name.IndexOfAny([System.IO.Path]::GetInvalidFileNameChars()) -ge 0 -or `
+            ($IsWindows -and ($Name.EndsWith('.') -or $Name.EndsWith(' ')))) {
+            Set-ItResult -Skipped -Because "'${Name}' is not supported as directory name on this operating system."
+            return
+        }
+
+        # note processing of '\' should be allowed on posix systems but 
+        # is quite broken in many powershell cmdlets
+        if ($Name.Contains('\')) 
+        {
+            Set-ItResult -Skipped -Because "path elements with backslashes are not fully supported in PowerShell on this operating system."
+            return
+        }
+
+        $dirFullName = (Join-Path $TestDrive $Name) 
+        
+        $newDir = [System.IO.Directory]::CreateDirectory($dirFullName)
```

Comment:
```suggestion
        $newDir = New-Item -Type Directory -Name $dirFullName
```

Reply:
Actually, the reason I didn't do it this way is because `New-Item -Type Directory -Name 'example '` erroneously truncates the extra space on Linux/MacOS and it shoudn't.  I'll put in a separate issue to track that issue.